feat(rest): add OpenAPI documentation page

yanavasileva opened the pull request:
Related to CAM-11766

ThorbenLindhauer commented on content/reference/rest/openapi/_index.md:
```diff
@@ -0,0 +1,61 @@
+[OpenAPI Specification 3.0.2](https://github.com/OAI/OpenAPI-Specification/blob/3.0.2/versions/3.0.2.md) [*]
+allowing client generation in different languages.
+
+[*] _OpenAPI Specification defines a standard, language-agnostic interface to RESTful APIs which allows both humans and computers to discover and understand the capabilities of the service without access to source code, documentation, or through network traffic inspection._
```
```suggestion
[*] _The OpenAPI Specification defines a standard, language-agnostic interface to RESTful APIs which allows both humans and computers to discover and understand the capabilities of the service without access to source code, documentation, or through network traffic inspection._
```

ThorbenLindhauer commented on content/reference/rest/openapi/_index.md:
```diff
@@ -0,0 +1,61 @@
+
+# Coverage
+
+The documentation is not complete, and it contains only the basic endpoints that are needed for a minimal REST API usage:
```
Let's just list the supported endpoints. "minimal" and "basic" are opinions that readers should form themselves, if they agree :)

yanavasileva replied:
👍

ThorbenLindhauer commented on content/reference/rest/openapi/_index.md:
```diff
@@ -0,0 +1,61 @@
+
+[*] _OpenAPI Specification defines a standard, language-agnostic interface to RESTful APIs which allows both humans and computers to discover and understand the capabilities of the service without access to source code, documentation, or through network traffic inspection._
+
+# Coverage
```
An idea: we could move this section to the end so that the first section is how to use it.

ThorbenLindhauer commented on content/reference/rest/openapi/_index.md:
```diff
@@ -0,0 +1,61 @@
+
+# Format
+The documentation is shipped as a single `openapi.json` file archived in a jar artifact.
+Download the Camunda REST API artifact containing the OpenAPI documentation [here](https://app.camunda.com/nexus/repository/camunda-bpm/org/camunda/bpm/camunda-engine-rest-openapi/7.13.0-alpha4/camunda-engine-rest-openapi-7.13.0-alpha4.jar).
```
Is it possible that we can write the link such that it does not become outdated?

yanavasileva replied:
Is this one better: https://app.camunda.com/nexus/service/rest/repository/browse/camunda-bpm/org/camunda/bpm/camunda-engine-rest-openapi/

ThorbenLindhauer replied:
Yeah, I think so.

ThorbenLindhauer commented on content/reference/rest/openapi/_index.md:
```diff
@@ -0,0 +1,61 @@
+
+# Format
+The documentation is shipped as a single `openapi.json` file archived in a jar artifact.
+Download the Camunda REST API artifact containing the OpenAPI documentation [here](https://app.camunda.com/nexus/repository/camunda-bpm/org/camunda/bpm/camunda-engine-rest-openapi/7.13.0-alpha4/camunda-engine-rest-openapi-7.13.0-alpha4.jar).
```
I think it helps if we describe how to obtain the JSON from the JAR, i.e. explain that a zip archiver can be used to unpack JAR files.

ThorbenLindhauer commented on content/reference/rest/openapi/_index.md:
````diff
@@ -0,0 +1,61 @@
+
+```
+java -jar openapi-generator-cli-VERSION.jar generate -i $PATH_TO_OPENAPI/openapi.json -g csharp-netcore -o OUTPUT_DIR
+```
````
Is there a good tutorial on using this generator that we can link to? Many people will approach this section with the question "How do I generate a client in language X?". We should not explain this step-by-step in our docs, but if there's a good secondary resource, that would be very useful.

ThorbenLindhauer commented on content/reference/rest/openapi/_index.md:
```diff
@@ -0,0 +1,61 @@
+
+[*] _If you need to install Java Runtime Environment, you can [find the download from Oracle here](https://www.oracle.com/java/technologies/javase-downloads.html)._
+
+# Try it out
```
If we explain the use cases in the introduction, this title should be aligned with it.

ThorbenLindhauer commented on content/reference/rest/openapi/_index.md:
```diff
@@ -0,0 +1,61 @@
+Instead of client generation, you can use one of the OpenAPI editors to play around with the REST API endpoints.
+
+For example, go to [Swagger Editor](https://editor.swagger.io/) and paste the content of the openapi.json on the left-hand side in the editor.
+Start a Process engine with enabled cross-origin requests, and you will be able to execute requests from the editor.
```
Maybe we can add another subsection on how to import the requests in Postman. And if there is a good secondary resource that explains that, then the content can be essentially just a link to that.

yanavasileva commented on content/reference/rest/openapi/_index.md:
```diff
@@ -0,0 +1,61 @@
+---
+
+title: "OpenAPI documentation"
```
👍

yanavasileva commented on content/reference/rest/openapi/_index.md:
```diff
@@ -0,0 +1,61 @@
+---
+
+
+Camunda REST API has partial OpenAPI documentation that follows
+[OpenAPI Specification 3.0.2](https://github.com/OAI/OpenAPI-Specification/blob/3.0.2/versions/3.0.2.md) [*]
+allowing client generation in different languages.
+
+[*] _OpenAPI Specification defines a standard, language-agnostic interface to RESTful APIs which allows both humans and computers to discover and understand the capabilities of the service without access to source code, documentation, or through network traffic inspection._
```
👍

yanavasileva commented:
I merged the `Format` section the introduction at the beginning (we can leave it as a titled section) and moved `Coverage` to the bottom of the page. Added links to the steps of client generation and Postman guides.

ThorbenLindhauer commented:
Thanks for making the requested changes. I made some small wording changes and added the maven coordinates. Then I merged the PR via https://github.com/camunda/camunda-docs-manual/commit/d8714d5e40991cb73e582746da1de18dcd71a685.
